scripts/check_deps.sh

```diff
@@ -35,20 +35,33 @@ missing_dep() {
 }
 
 GO_DEPS=(
-    "$GO_BIN/stringer"
-    "$GO_BIN/msgp"
-    "$GO_BIN/golangci-lint"
+    "msgp"
+    "golangci-lint"
+    "oapi-codegen"
+    "swagger"
 )
 
+check_go_binary_version() {
+  binary_name=$1
+  expected_version=$(grep "$binary_name" scripts/buildtools/versions | awk '{print $2}')
+  actual_version=$(go version -m "$GO_BIN/$binary_name" | awk 'NR==3 {print $3}')
+
+  if [ "$expected_version" != "$actual_version" ]; then
```

Comment:
```suggestion
  if [ "$(printf '%s\n' "$expected_version" "$actual_version" | sort -V | head -n1)" != "$expected_version" ]; then
```

Reply:
what does this do? it looks like it's calling printf with one pattern and two args?

Reply:
oh, you're trying to allow more recent stuff?